fili-core/src/main/java/com/yahoo/bard/webservice/druid/client/impl/HeaderNestingJsonBuilderStrategy.java

```diff
@@ -48,6 +49,16 @@ public JsonNode apply(Response response) {
                     DruidJsonResponseContentKeys.STATUS_CODE.getName(),
                     mappingJsonFactory.createParser(String.valueOf(response.getStatusCode())).readValueAsTree()
             );
+            if (CacheFeatureFlag.ETAG.isOn()) {
+                objectNode.set(
+                        DruidJsonResponseContentKeys.ETAG.getName(),
+                        mappingJsonFactory
+                                .createParser(
+                                        response.getHeader(DruidJsonResponseContentKeys.ETAG.getName())
```

Comment:
https://github.com/yahoo/fili/issues/255
Look at the headers for Cache hit and Cache miss responses.
We don't always get the Etag header back, and If-None-Match is not the response header for etags.